HealthVault/Classes/Caching/Private/MHVThingCache.m

```diff
@@ -596,16 +684,261 @@ - (void)performSyncCompletionsWithSyncedCount:(NSInteger)syncedItemCount error:(
+- (MHVAsyncTask *)taskToExecutePendingMethod:(MHVPendingMethod *)pendingMethod
+{
+    return [[MHVAsyncTask alloc] initWithIndeterminateBlock:^(id input, void (^finish)(id), void (^cancel)(id))
+    {
+        [self.connection executeHttpServiceOperation:pendingMethod
+                                          completion:^(MHVServiceResponse * _Nullable response, NSError * _Nullable error)
+        {
+            if (error)
+            {
+                cancel([MHVAsyncTaskResult withError:error]);
+            }
+            else
+            {
+                [self.database deletePendingMethods:@[pendingMethod]
+                                         completion:^(NSError * _Nullable error)
+                {
+                    if (error)
+                    {
+                        cancel([MHVAsyncTaskResult withError:error]);
+                    }
+                    else
+                    {
+                        finish([MHVAsyncTaskResult withResult:nil]);
+                    }
+                }];
+            }
+        }];
+    }];
+}
+
+// Task to get the record operations since the last sync. Will CANCEL with an MHVAsyncTaskResult<NSError *> or FINISH with MHVAsyncTaskResult<NSNumber *>.
+- (MHVAsyncTask *)taskForRecordOperationsWithRecordId:(NSString *)recordId
+{
+    return [[MHVAsyncTask alloc] initWithIndeterminateBlock:^(MHVAsyncTaskResult *input, void (^finish)(id), void (^cancel)(id))
+    {
+        id<MHVCacheStatusProtocol> status = input.result;
+        
+        if (!status)
+        {
+            cancel([MHVAsyncTaskResult withError:[NSError MHVCacheError:@"Could not determine the status of the cache database."]]);
+            return;
+        }
+        
+        // If the lastCacheConsistencyDate value is set and the time between now and the lastCompletedSyncDate is < syncIntervalSeconds don't sync.
+        if (status.lastCacheConsistencyDate && fabs([status.lastCompletedSyncDate timeIntervalSinceNow]) < self.cacheConfiguration.syncIntervalSeconds)
+        {
+            MHVLOG(@"ThingCache: Record is up to date, synced %li seconds ago", (long)fabs([status.lastCompletedSyncDate timeIntervalSinceNow]));
+            
+            finish([MHVAsyncTaskResult withResult:nil]);
+            return;
+        }
+        
+        [self.connection.thingClient getRecordOperations:status.newestCacheSequenceNumber
+                                                recordId:[[NSUUID alloc] initWithUUIDString:recordId]
+                                              completion:^(MHVGetRecordOperationsResult * _Nullable result, NSError * _Nullable error)
+        {
+            if (error)
+            {
+                cancel([MHVAsyncTaskResult withError:error]);
+            }
+            else
+            {
+                if (result.operations.count < 1)
+                {
+                    // No record operations sync the last sync
+                    finish([MHVAsyncTaskResult withResult:nil]);
```

Comment:
@{ kCacheStatusKey : status }
otherwise looks like below it won't have the status object and will reset the sequenceNumber to 0